### PR TITLE
[Doc] Update math module

### PR DESCRIPTION
**Pull request by Olinaaaloompa**
Issue: #

### Brief Summary
Editorial updates

### CODE REVIEW

File: docs/lang/articles/math/math_module.md [hunk 1]
```diff
@@ -49,34 +47,33 @@ def test():
...
 :::note
 
-Difference between Taichi's math module and Python's built-in math module:
+Taichi's math module has a large overlap with Python's built-in math module. But you should follow a few extra fules when using Taichi's math module:
 
-- Functions in Taichi's math module must be called from within the Taichi scope.
-- Functions in Taichi's math module also accept vectors/matrices as arguments.
-- The precision of a function in Taichi's math module depends on the value of `default_fp` and backend specified in the `ti.init()` method call.
++ Functions in Taichi's math module *must* be called from within the Taichi scope.
++ Functions in Taichi's math module also accept vectors and matrices as arguments.
```
Comment (Vissidarte-Herman) [inline docs/lang/articles/math/math_module.md:53]:
```suggestion
+ Functions in Taichi's math module also accept vectors or matrices as arguments.
```

File: docs/lang/articles/math/math_module.md [hunk 2]
```diff
@@ -49,34 +47,33 @@ def test():
 
 :::note
 
-Difference between Taichi's math module and Python's built-in math module:
+Taichi's math module has a large overlap with Python's built-in math module. But you should follow a few extra fules when using Taichi's math module:
```
Comment (Vissidarte-Herman) [inline docs/lang/articles/math/math_module.md:50]:
```suggestion
Taichi's math module has a large overlap with Python's built-in math module. But you should follow a few extra rules when using Taichi's math module:
```

File: docs/lang/articles/math/math_module.md [hunk 3]
```diff
@@ -8,15 +8,13 @@ sidebar_position: 1
...
++ Small vector and matrix types that are analogous to those in the [OpenGL shading language](https://www.khronos.org/opengl/wiki/OpenGL_Shading_Language) (GLSL).
 + Some GLSL-standard functions.
 + Complex number operations of 2D vectors.
 
-
-
 ## Mathematical functions
 
-Taichi's `math` module has a large overlap with Python's built-in `math` module. You need to call the functions in the Taichi scope. For example:
+You should call the mathematical functions supported by Taichi's `math` module from within the Taichi scope. For example:
```
Comment (Vissidarte-Herman) [inline docs/lang/articles/math/math_module.md:17]:
```suggestion
You should call the mathematical functions provided by Taichi's `math` module from within the Taichi scope. For example:
```

**Review by Vissidarte-Herman — CHANGES_REQUESTED** [review]

File: docs/lang/articles/math/math_module.md [hunk 4]
```diff
@@ -104,23 +101,24 @@ u = v.wzyx  # vec4(4, 3, 2, 1)
 u = v.rraa  # vec4(1, 1, 2, 2)
 ```
 
-### Relationship between `ti.Vector`, `ti.types.vector` and `ti.math.vec3`
+### Differentiate `ti.Vector`, `ti.types.vector` and `ti.math.vec3`
```
Comment (Vissidarte-Herman) [inline docs/lang/articles/math/math_module.md:104]:
```suggestion
### Relations between `ti.Vector`, `ti.types.vector` and `ti.math.vec3`
```
Differentiate is not the right word.

**Review by Vissidarte-Herman — CHANGES_REQUESTED** [review]

File: docs/lang/articles/math/math_module.md [hunk 5]
```diff
@@ -104,23 +101,24 @@ u = v.wzyx  # vec4(4, 3, 2, 1)
...
++ `ti.types.vector` is a function that accepts an integer and a primitive type and returns a vector type. For example: `vec5f = ti.types.vector(5, float)`. `vec5f` can then be used to instantiate 5D vectors or annotate data types of function arguments and struct members:
+
     ```python
     @ti.kernel
     def test(v: vec5f):
         print(v.xyz)
     ```
-    Unlike `ti.Vector`, whose input data must be a 1D array, vector types created by `ti.types.vector()` have more flexible ways to initialize, as explained above.
-+ `ti.math.vec3` is simply created by `vec3 = ti.types.vector(3, float)`.
+    Unlike `ti.Vector`, whose input data must be a 1D array, `ti.types.vector()` creates vector types that allow more flexible initiation rules, as explained above.
```
Comment (Vissidarte-Herman) [inline docs/lang/articles/math/math_module.md:114]: The original is grammatically correct. Please could I ask you to revert this change?

**Review by Vissidarte-Herman — CHANGES_REQUESTED** [review]

File: docs/lang/articles/math/math_module.md [hunk 6]
```diff
@@ -49,34 +47,33 @@ def test():
...
 
 ```python
 vec2 = ti.types.vector(2, float)
 ```
 
-The number of precision bits of such a type is determined by `default_fp` or `default_ip` in the `ti.init()` method call. For example, if `ti.init(default_fp=ti.f64)` is called, then `vec2/vec3/vec4` and `mat2/mat3/mat4` will all have 64-bit floating-point precision in the Taichi scope.
+The number of precision bits of such a type is determined by `default_fp` or `default_ip` in the `ti.init()` method call. For example, if `ti.init(default_fp=ti.f64)` is called, then `vec2/vec3/vec4` and `mat2/mat3/mat4` defined in the Taichi scope all have 64-bit floating-point precision.
 
-These types can be used to instantiate vectors and matrices or annotate the data types of function arguments and struct members. See [type system](../type_system/type.md) for more information. Here we emphasize that they have very flexible initialization routines:
+You can use these types to instantiate vectors/matrices or annotate the data types of function arguments and struct members. See the [Type System](../type_system/type.md) for more information. Note that flexible initialization rules apply:
```
[the remark this answers is not in the document]
Comment (neozhaoliang) [reply docs/lang/articles/math/math_module.md:76]: It should mean there are various ways to instantiate a vector. Very flexible.

File: docs/lang/articles/math/math_module.md [hunk 7]
```diff
@@ -104,23 +101,24 @@ u = v.wzyx  # vec4(4, 3, 2, 1)
...
-+ `ti.math.vec3` is simply created by `vec3 = ti.types.vector(3, float)`.
+    Unlike `ti.Vector`, whose input data must be a 1D array, `ti.types.vector()` creates vector types that allow more flexible initiation rules, as explained above.
++ `ti.math.vec3` is created by `vec3 = ti.types.vector(3, float)`.
 
 
 ## GLSL-standard functions
 
 
-Taichi's math module also supports a few [GLSL standard functions](https://registry.khronos.org/OpenGL-Refpages/gl4/index.php). These functions follow the GLSL standard, except that they accept arbitrary vectors and matrices as arguments and operate on them element-wise. For example:
+Taichi's math module also supports a few [GLSL standard functions](https://registry.khronos.org/OpenGL-Refpages/gl4/index.php). These functions follow the GLSL standard, except that they accept arbitrary vectors and matrices as arguments and perform element-wise operations. For example:
```
Comment (Vissidarte-Herman) [inline docs/lang/articles/math/math_module.md:121]: The original is good. Could I ask you to use the original?
Your version is unclear because it didn't specify the target of these operations: The matrices and vectors.

**Review by Vissidarte-Herman — CHANGES_REQUESTED** [review]

File: docs/lang/articles/math/math_module.md [hunk 8]
```diff
@@ -93,7 +90,7 @@ u = vec4([1, 2], [3, 4])
 u = vec4(v, 4.0)
 ```
 
-Another important feature of vector types created by `ti.types.vector()` is that they support **vector swizzling** just as GLSL vectors do. This means you can use `xyzw`, `rgba`, `stpq` to access their elements with indices &le; four:
+Similar to GLSL vectors, the vector types created by `ti.types.vector()` also support **vector swizzling**. This means you can use `xyzw`, `rgba`, or `stpq` to access their elements with no more than four indices:
```
Comment (Vissidarte-Herman) [inline docs/lang/articles/math/math_module.md:93]: If you use "similar" then "also" is redundant. This means -> This means that: Use that to lead an appositive clause. 

Please use the origin. I edited this line before.

**Review by Vissidarte-Herman — CHANGES_REQUESTED** [review]

File: docs/lang/articles/math/math_module.md [hunk 9]
```diff
@@ -143,9 +141,9 @@ Texture support in Taichi is implemented in the `ti.types.texture_types` module.
 
 ## Complex number operations
 
-Taichi's math module also supports basic complex arithmetic in the form of 2D vectors.
+Taichi's math module also supports basic complex number operations via 2D vectors.
```
Comment (Vissidarte-Herman) [inline docs/lang/articles/math/math_module.md:144]:
```suggestion
Taichi's math module also supports basic complex arithmetic operations on 2D vectors.
```
"via" is incorrect.

File: docs/lang/articles/math/math_module.md [hunk 10]
```diff
@@ -160,7 +158,7 @@ def test():
     w = tm.cdiv(x, y)  #  vec2(2, 0) = 2+0j
 ```
 
-You can also compute the power, logarithm and exponetial of a complex number:
+You can also compute the power, logarithm and exponential of a complex number:
```
Comment (Vissidarte-Herman) [inline docs/lang/articles/math/math_module.md:161]: a, b, and c
```suggestion
You can also compute the power, logarithm, and exponential of a complex number:
```

File: docs/lang/articles/math/math_module.md [hunk 11]
```diff
@@ -143,9 +141,9 @@ Texture support in Taichi is implemented in the `ti.types.texture_types` module.
 
 ## Complex number operations
 
-Taichi's math module also supports basic complex arithmetic in the form of 2D vectors.
+Taichi's math module also supports basic complex number operations via 2D vectors.
 
-You can use a 2D vector of type `ti.math.vec2` to represent a complex number, the addtion/subtraction of complex numbers are just addtion/subtraction of 2D vectors, and multiplication and division can be performed by calling `ti.math.cmul` and `ti.math.cdiv`:
+You can use a 2D vector of type `ti.math.vec2` to represent a complex number. In this way, the addition and subtraction of complex numbers are equivalent to the addtion and subtraction of 2D vectors, and the multiplication and division of complex numbers can be performed by calling `ti.math.cmul` and `ti.math.cdiv`:
```
Comment (Vissidarte-Herman) [inline docs/lang/articles/math/math_module.md:146]:
```suggestion
You can use a 2D vector of type `ti.math.vec2` to represent a complex number. In this way, additions and subtractions of complex numbers come in the form of 2D vector additions and subtractions. You can call  `ti.math.cmul()` and `ti.math.cdiv()` to conduct multiplication and division of complex numbers:
```
The second "addtion" is incorrect.

**Review by Vissidarte-Herman — CHANGES_REQUESTED** [review]

File: docs/lang/articles/math/math_module.md [hunk 12]
```diff
@@ -8,15 +8,13 @@ sidebar_position: 1
...
++ Small vector and matrix types that are analogous to those in the [OpenGL shading language](https://www.khronos.org/opengl/wiki/OpenGL_Shading_Language) (GLSL).
 + Some GLSL-standard functions.
 + Complex number operations of 2D vectors.
 
-
-
 ## Mathematical functions
 
-Taichi's `math` module has a large overlap with Python's built-in `math` module. You need to call the functions in the Taichi scope. For example:
+You should call the mathematical functions provided by Taichi's `math` module from within the Taichi scope. For example:
```
Comment (Vissidarte-Herman) [inline docs/lang/articles/math/math_module.md:17]:
```suggestion
You must call the mathematical functions provided by Taichi's `math` module from within the Taichi scope. For example:
```
Style thing: use "must" for "should"

File: docs/lang/articles/math/math_module.md [hunk 13]
```diff
@@ -33,7 +31,7 @@ def test():
     ...
 ```
 
-These functions also accept vectors and matrices as arguments, and they apply to them element-wise:
+These functions also accept vectors and matrices as arguments and conduct element-wise operations:
```
Comment (Vissidarte-Herman) [inline docs/lang/articles/math/math_module.md:34]:
```suggestion
These functions also take vectors and matrices as arguments and operate on them element-wise:
```
"take xxx as arguments" fixed usage.

**Review by Vissidarte-Herman — CHANGES_REQUESTED** [review]

File: docs/lang/articles/math/math_module.md [hunk 14]
```diff
@@ -49,34 +47,33 @@ def test():
 
 :::note
 
-Difference between Taichi's math module and Python's built-in math module:
+Taichi's math module has a large overlap with Python's built-in math module. But you should follow a few extra rules when using Taichi's math module:
 
-- Functions in Taichi's math module must be called from within the Taichi scope.
-- Functions in Taichi's math module also accept vectors/matrices as arguments.
-- The precision of a function in Taichi's math module depends on the value of `default_fp` and backend specified in the `ti.init()` method call.
++ Functions in Taichi's math module *must* be called from within the Taichi scope.
```
Comment (Vissidarte-Herman) [inline docs/lang/articles/math/math_module.md:52]: Active voice. Use active voice as much as possible. 

```suggestion
- You must call the functions provided by Taichi's math moduled from within the Taichi scope.
```
use "-" for bullet. Please update the entire document.

**Review by Vissidarte-Herman — CHANGES_REQUESTED** [review]

File: docs/lang/articles/math/math_module.md [hunk 15]
```diff
@@ -49,34 +47,33 @@ def test():
...
++ Functions in Taichi's math module also accept vectors or matrices as arguments.
```
Comment (Vissidarte-Herman) [inline docs/lang/articles/math/math_module.md:53]:
```suggestion
- Functions in Taichi's math module also take vectors or matrices as arguments.
```

File: docs/lang/articles/math/math_module.md [hunk 16]
```diff
@@ -49,34 +47,33 @@ def test():
...
++ The precision of a function in Taichi's math module depends on the value of `default_fp` and the backend specified in the `ti.init()` method call.
```
Comment (Vissidarte-Herman) [inline docs/lang/articles/math/math_module.md:54]:
```suggestion
- The precision of a function in Taichi's math module depends on the value of `default_fp` and the backend specified in `ti.init()`.
```

**Review by Vissidarte-Herman — CHANGES_REQUESTED** [review]

File: docs/lang/articles/math/math_module.md [hunk 17]
```diff
@@ -104,17 +101,18 @@ u = v.wzyx  # vec4(4, 3, 2, 1)
 u = v.rraa  # vec4(1, 1, 2, 2)
 ```
 
-### Relationship between `ti.Vector`, `ti.types.vector` and `ti.math.vec3`
+### Relations between `ti.Vector`, `ti.types.vector` and `ti.math.vec3`
 
 + `ti.Vector` is a function that accepts a 1D array and returns a matrix instance that has only one column. For example, `ti.Vector([1, 2, 3, 4, 5])`.
-+ `ti.types.vector` is a function that accepts an integer and a primitive type, and returns a vector type. For example: `vec5f = ti.types.vector(5, float)`. `vec5f` can then be used to instantiate 5D vectors or annotate data types in function arguments and struct members:
++ `ti.types.vector` is a function that accepts an integer and a primitive type and returns a vector type. For example: `vec5f = ti.types.vector(5, float)`. `vec5f` can then be used to instantiate 5D vectors or annotate data types of function arguments and struct members:
```
Comment (Vissidarte-Herman) [inline docs/lang/articles/math/math_module.md]:
```suggestion
+ `ti.types.vector()` is a function that accepts an integer and a primitive type and returns a vector type. For example: `vec5f = ti.types.vector(5, float)`. You can then use `vec5f` to instantiate a 5D vector or annotate data types for function arguments and struct members:
```
The best-darn edit in this line is to change the preposition "of" to "for". Brrrrrrrrrravo!!

**Review by Vissidarte-Herman — CHANGES_REQUESTED** [review]

File: docs/lang/articles/math/math_module.md [hunk 18]
```diff
@@ -49,34 +47,33 @@ def test():
...
 
 ```python
 vec2 = ti.types.vector(2, float)
 ```
 
-The number of precision bits of such a type is determined by `default_fp` or `default_ip` in the `ti.init()` method call. For example, if `ti.init(default_fp=ti.f64)` is called, then `vec2/vec3/vec4` and `mat2/mat3/mat4` will all have 64-bit floating-point precision in the Taichi scope.
+The number of precision bits of such a type is determined by `default_fp` or `default_ip` in the `ti.init()` method call. For example, if `ti.init(default_fp=ti.f64)` is called, then `vec2/vec3/vec4` and `mat2/mat3/mat4` defined in the Taichi scope all have 64-bit floating-point precision.
 
-These types can be used to instantiate vectors and matrices or annotate the data types of function arguments and struct members. See [type system](../type_system/type.md) for more information. Here we emphasize that they have very flexible initialization routines:
+You can use these types to instantiate vectors/matrices or annotate the data types of function arguments and struct members. See the [Type System](../type_system/type.md) for more information. Here we emphasize that they have very flexible initialization routines:
```
Comment (Vissidarte-Herman) [inline docs/lang/articles/math/math_module.md:76]:
```suggestion
You can use these types to instantiate vectors/matrices or annotate data types for function arguments and struct members. See the [Type System](../type_system/type.md) for more information. Here we emphasize that they have very flexible initialization routines:
```

**Review by Vissidarte-Herman — CHANGES_REQUESTED** [review]

File: docs/lang/articles/math/math_module.md [hunk 19]
```diff
@@ -49,34 +47,33 @@ def test():
...
-
-These vector/matrix types are created with the two template functions `ti.types.vector()` and `ti.types.matrix()`. For example, `vec2` is defined in the following way:
+To create one of the vector/matrix types above, use the template `ti.types.vector()` or `ti.types.matrix()`. For example, `vec2` is defined in the following way:
 
 ```python
 vec2 = ti.types.vector(2, float)
 ```
 
-The number of precision bits of such a type is determined by `default_fp` or `default_ip` in the `ti.init()` method call. For example, if `ti.init(default_fp=ti.f64)` is called, then `vec2/vec3/vec4` and `mat2/mat3/mat4` will all have 64-bit floating-point precision in the Taichi scope.
+The number of precision bits of such a type is determined by `default_fp` or `default_ip` in the `ti.init()` method call. For example, if `ti.init(default_fp=ti.f64)` is called, then `vec2/vec3/vec4` and `mat2/mat3/mat4` defined in the Taichi scope all have 64-bit floating-point precision.
```
Comment (Vissidarte-Herman) [inline docs/lang/articles/math/math_module.md:74]:
```suggestion
The number of precision bits of such a type is determined by `default_fp` or `default_ip` in the `ti.init()` method call. For example, if `ti.init(default_fp=ti.f64)` is called, then `vec2/vec3/vec4` and `mat2/mat3/mat4` defined in the Taichi scope all have a 64-bit floating-point precision.
```

File: docs/lang/articles/math/math_module.md [hunk 20]
```diff
@@ -49,34 +47,33 @@ def test():
...
-+ `uvec2/uvec3/uvec4` for 2D/3D/4D unsigned integer vector types.
-+ `mat2/mat3/mat4` for 2D/3D/4D floating-point square matrix types.
+- `vec2/vec3/vec4` for 2D/3D/4D floating-point vector types.
+- `ivec2/ivec3/ivec4` for 2D/3D/4D integer vector types.
+- `uvec2/uvec3/uvec4` for 2D/3D/4D unsigned integer vector types.
+- `mat2/mat3/mat4` for 2D/3D/4D floating-point square matrix types.
 
-
-These vector/matrix types are created with the two template functions `ti.types.vector()` and `ti.types.matrix()`. For example, `vec2` is defined in the following way:
+To create one of the vector/matrix types above, use the template `ti.types.vector()` or `ti.types.matrix()`. For example, `vec2` is defined in the following way:
```
Comment (Vissidarte-Herman) [inline docs/lang/articles/math/math_module.md:68]:
```suggestion
To create one of the vector/matrix types above, use template function `ti.types.vector()` or `ti.types.matrix()`. For example, `vec2` is defined as follows:
```

**Review by Vissidarte-Herman — CHANGES_REQUESTED** [review]

File: docs/lang/articles/math/math_module.md [hunk 21]
```diff
@@ -7,16 +7,14 @@ sidebar_position: 1
...
-+ Commonly-used mathematical functions that are analogous to those in Python's built-in `math` module.
-+ Small vector and matrix types that are analogous to those in the [OpenGL shading language](https://www.khronos.org/opengl/wiki/OpenGL_Shading_Language)(GLSL).
-+ Some GLSL-standard functions.
-+ Complex number operations of 2D vectors.
-
-
+- Commonly-used mathematical functions that are analogous to those in Python's built-in `math` module.
+- Small vector and matrix types that are analogous to those in the [OpenGL shading language](https://www.khronos.org/opengl/wiki/OpenGL_Shading_Language) (GLSL).
+- Some GLSL-standard functions.
+- Complex number operations of 2D vectors.
```
Comment (Vissidarte-Herman) [inline docs/lang/articles/math/math_module.md:13]:
```suggestion
- Complex number operations in the form of 2D vectors.
```

**Review by Vissidarte-Herman — CHANGES_REQUESTED** [review]

File: docs/lang/articles/math/math_module.md [hunk 22]
```diff
@@ -49,34 +47,33 @@ def test():
 
 :::note
 
-Difference between Taichi's math module and Python's built-in math module:
+Taichi's math module has a large overlap with Python's built-in math module. But you should follow a few extra rules when using Taichi's math module:
 
-- Functions in Taichi's math module must be called from within the Taichi scope.
-- Functions in Taichi's math module also accept vectors/matrices as arguments.
-- The precision of a function in Taichi's math module depends on the value of `default_fp` and backend specified in the `ti.init()` method call.
+- You must call the functions provided by Taichi's math moduled from within the Taichi scope.
```
Comment (Vissidarte-Herman) [inline docs/lang/articles/math/math_module.md:52]: moduled?

**Review by Vissidarte-Herman — CHANGES_REQUESTED** [review]

File: docs/lang/articles/math/math_module.md [hunk 23]
```diff
@@ -49,34 +47,33 @@ def test():
 
 :::note
 
-Difference between Taichi's math module and Python's built-in math module:
+Taichi's math module has a large overlap with Python's built-in math module. But you should follow a few extra rules when using Taichi's math module:
```
Comment (Vissidarte-Herman) [inline docs/lang/articles/math/math_module.md:50]:
```suggestion
Taichi's math module overlaps to a large extent with Python's built-in math module. Ensure that you follow a few extra rules when using Taichi's math module:
```

Comment (Vissidarte-Herman) [reply docs/lang/articles/math/math_module.md]: Remember the noun form is ALWAYS wordier than the original verb form.

File: docs/lang/articles/math/math_module.md [hunk 24]
```diff
@@ -49,34 +47,33 @@ def test():
...
 
-- Functions in Taichi's math module must be called from within the Taichi scope.
-- Functions in Taichi's math module also accept vectors/matrices as arguments.
-- The precision of a function in Taichi's math module depends on the value of `default_fp` and backend specified in the `ti.init()` method call.
+- You must call the functions provided by Taichi's math module from within the Taichi scope.
+- Functions in Taichi's math module also accept vectors or matrices as arguments.
```
Comment (Vissidarte-Herman) [inline docs/lang/articles/math/math_module.md:53]:
```suggestion
- Functions in Taichi's math module also take vectors or matrices as arguments.
```

File: docs/lang/articles/math/math_module.md [hunk 25]
```diff
@@ -49,34 +47,33 @@ def test():
...
-+ `uvec2/uvec3/uvec4` for 2D/3D/4D unsigned integer vector types.
-+ `mat2/mat3/mat4` for 2D/3D/4D floating-point square matrix types.
+- `vec2/vec3/vec4` for 2D/3D/4D floating-point vector types.
+- `ivec2/ivec3/ivec4` for 2D/3D/4D integer vector types.
+- `uvec2/uvec3/uvec4` for 2D/3D/4D unsigned integer vector types.
+- `mat2/mat3/mat4` for 2D/3D/4D floating-point square matrix types.
 
-
-These vector/matrix types are created with the two template functions `ti.types.vector()` and `ti.types.matrix()`. For example, `vec2` is defined in the following way:
+To create one of the vector/matrix types above, use template function `ti.types.vector()` or `ti.types.matrix()`. For example, `vec2` is defined as follows:
```
Comment (Vissidarte-Herman) [inline docs/lang/articles/math/math_module.md:68]:
```suggestion
To create one of the vector/matrix types above, use template function `ti.types.vector()` or `ti.types.matrix()`. For example, `vec2` is defined as follows:
```
Do you know why I omit "the" here? Because it's not definite.

File: docs/lang/articles/math/math_module.md [hunk 26]
```diff
@@ -49,34 +47,33 @@ def test():
...
 
-Taichi's math module supplies a few small vector and matrix types:
+Taichi's math module provides a few small vector and matrix types:
 
 
-+ `vec2/vec3/vec4` for 2D/3D/4D floating-point vector types.
-+ `ivec2/ivec3/ivec4` for 2D/3D/4D integer vector types.
-+ `uvec2/uvec3/uvec4` for 2D/3D/4D unsigned integer vector types.
-+ `mat2/mat3/mat4` for 2D/3D/4D floating-point square matrix types.
+- `vec2/vec3/vec4` for 2D/3D/4D floating-point vector types.
```
Comment (Vissidarte-Herman) [inline docs/lang/articles/math/math_module.md:63]:
```suggestion
- `vec2/vec3/vec4`: 2D/3D/4D floating-point vector types.
```
Same for the other three.

**Review by Vissidarte-Herman — APPROVED** [review]

File: docs/lang/articles/math/math_module.md [hunk 27]
```diff
@@ -49,34 +47,33 @@ def test():
...
 
-Difference between Taichi's math module and Python's built-in math module:
+Taichi's math module overlaps to a large extent with Python's built-in math module. Ensure that you follow a few extra rules when using Taichi's math module:
 
-- Functions in Taichi's math module must be called from within the Taichi scope.
-- Functions in Taichi's math module also accept vectors/matrices as arguments.
-- The precision of a function in Taichi's math module depends on the value of `default_fp` and backend specified in the `ti.init()` method call.
+- You must call the functions provided by Taichi's math module from within the Taichi scope.
+- Functions in Taichi's math module also take vectors or matrices as arguments.
+- The precision of a function in Taichi's math module depends on the value of `default_fp` and the backend specified in the `ti.init()` method call.
```
Comment (Vissidarte-Herman) [inline docs/lang/articles/math/math_module.md:54]:
```suggestion
- The precision of a function in Taichi's math module depends on the settings of `default_fp` and `arch` (backend) in `ti.init()`.
```

**Review by Vissidarte-Herman — CHANGES_REQUESTED** [review]

File: docs/lang/articles/math/math_module.md [hunk 28]
```diff
@@ -143,9 +141,9 @@ Texture support in Taichi is implemented in the `ti.types.texture_types` module.
 
 ## Complex number operations
 
-Taichi's math module also supports basic complex arithmetic in the form of 2D vectors.
+Taichi's math module also supports basic complex arithmetic operations on 2D vectors.
 
-You can use a 2D vector of type `ti.math.vec2` to represent a complex number, the addtion/subtraction of complex numbers are just addtion/subtraction of 2D vectors, and multiplication and division can be performed by calling `ti.math.cmul` and `ti.math.cdiv`:
+You can use a 2D vector of type `ti.math.vec2` to represent a complex number. In this way, additions and subtractions of complex numbers come in the form of 2D vector additions and subtractions. You can call  `ti.math.cmul()` and `ti.math.cdiv()` to conduct multiplication and division of complex numbers:
```
Comment (Vissidarte-Herman) [inline docs/lang/articles/math/math_module.md:146]:
```suggestion
You can use a 2D vector of type `ti.math.vec2` to represent a complex number. In this way, additions and subtractions of complex numbers come in the form of 2D vector additions and subtractions. You can call `ti.math.cmul()` and `ti.math.cdiv()` to conduct multiplication and division of complex numbers:
```

**Review by Vissidarte-Herman — CHANGES_REQUESTED** [review]

**Review by Vissidarte-Herman — APPROVED** [review]
